logging/unit_tests/test_client.py

```diff
@@ -519,22 +539,40 @@ def test_list_metrics_with_paging(self):
             'filter': self.FILTER,
             'description': self.DESCRIPTION,
         }]
-        client = self._makeOne(project=PROJECT, credentials=_Credentials())
-        api = client._metrics_api = _DummyMetricsAPI()
-        api._list_metrics_response = METRICS, None
+        client = self._makeOne(project=PROJECT, credentials=_Credentials(),
+                               use_gax=False)
+        returned = {
+            'metrics': METRICS,
+        }
+        client.connection = _Connection(returned)
```

Comment:
*This* version tests the "list all items" mode in both test cases, but not the "page-at-a-time" mode.